docs(guides): Add practical 'what to do next' guides for key Sentry features (#16410)
getsentry/sentry-docs: docs/guides/issues-errors.mdx

---
title: "What to Watch"
sidebar_order: 5
description: "Practical guidance on what errors to catch, how to search issues, and when to set alerts."
---

Your Sentry SDK is sending errors to [Sentry Issues](/product/issues/) out of the box. Now what? This guide covers the high-value error patterns that help you catch problems before they impact users and fix issues faster.

> ## The Pattern

Reviewer: Hm, maybe we can better outline that, or just say that as the subheader? I didn't get it just from the header itself.

> Sentry automatically captures unhandled errors. For handled errors, use `Sentry.captureException()` to add context:

Reviewer: ```suggestion
Sentry automatically captures unhandled errors. For handled errors, use `Sentry.captureException()` and add context:
```

```javascript
try {
  await processOrder(order);
} catch (error) {
  Sentry.captureException(error, {
    tags: {
      order_id: order.id,
      payment_method: order.paymentMethod
    },
    level: "error"
  });
  throw error;
}
```

**Levels:** `fatal`, `error`, `warning`, `info`, `debug`

> **Context:** Tags, user info, and breadcrumbs help you debug faster. Tags are searchable, so use them for high-cardinality data like IDs, regions, and feature flags.

Reviewer: I thought we generally do not want to reference breadcrumbs?  We should mention logs and replays instead, yeah?

> Every error in Sentry is automatically trace-connected. Click the trace ID to see the full [trace view](/concepts/key-terms/tracing/trace-view/) and understand what led to the error.

Reviewer: I think we need to wait if we're showing performance stuff. We're about to release a new UX on all these dashboards. 

Meta comment on that too - this surface is super hard to find in product right now, and the trace view doc is quite dated. I purposefully have not updated it because it's rolled into the insights > dashboards updates I'm doing. I'm wondering if we can direct link to the product page for now instead?

## Where to Look

> Start with these five views to catch critical issues.

Reviewer: ```suggestion
Start by creating these five views to catch the most critical issues.
```
I'm thinking if we feel like these are really valuable, we should suggest people save them to use them over and over again.

### 1. Unresolved High-Volume Issues

High event counts mean either many users are affected or one user is stuck in a loop. Both need immediate attention.

> **In Issues:** Search for `is:unresolved` (applied by default) and use the sort dropdown (defaults to "Last Seen") to sort by **Events** 

Reviewer: I realized these are all issue filters, so bumped the link up to the top. It still might be useful to add a link in each section where we say `In Issues`, though. What do you think?

- New error types (TypeError, ReferenceError) indicating breaking changes

Learn more about [Release Health](/product/releases/health/) to track error rates and crash-free sessions per release.

**Alert idea:** Trigger when a new issue is created, filtered to production environment and error/fatal level. This catches regressions immediately after deploy.

### 3. Errors by Environment

Production errors are critical. Staging errors help you catch problems before they reach users. Comparing environments helps you spot configuration issues.

**In Issues:** Search for `is:unresolved environment:production` or `environment:staging`

**What to look for:**
- Errors only in production (often config, API keys, or data issues)
- Errors only in staging (caught before deploy, needs fixing)
- Same error in both (systemic code issue)

Learn more about <PlatformLink to="/configuration/environments/">configuring environments</PlatformLink> to separate production and staging issues.

**Alert idea:** Trigger when a new issue is created, filtered to production environment. These often indicate config drift or environment-specific bugs.

### 4. Errors with High User Impact

> Some errors affect power users heavily. Others affect everyone rarely. Prioritize by unique user count, not just event count.

Reviewer 2: ```suggestion
Some errors heavily affect power users. Others affect everyone, but rarely. Prioritize by unique user count, not just event count.
```

Learn more about <PlatformLink to="/enriching-events/tags/">custom tags</PlatformLink> to mark user tiers, plans, or roles. Then search `is:unresolved user.tier:enterprise` to prioritize high-value users.

**Alert idea:** Trigger when unique user count exceeds a threshold in a time window (e.g., more than 50 users in 1 hour) to catch widespread issues.

### 5. User-Reported Issues

When users report problems, they often see an error. User feedback tells you which errors hurt the experience most, even if event counts are low.

**In Issues:** Navigate to **User Feedback** in the left sidebar to see reports submitted by users.

**What to look for:**
- User descriptions of what they were doing when the error occurred
- Patterns in feedback about specific workflows or features
- Issues that frustrate users even if they don't generate many events

Enable the [User Feedback Widget](/platforms/javascript/user-feedback/) to let users report problems directly when errors happen.

**Alert idea:** Trigger when a new issue is created, filtered to issue category equals feedback. This ensures you respond quickly when users report problems.

> ## Creating Issue Alerts

Reviewer: Yes to all this. Just know that this UX is changing, with Monitors and Alerts in LA.

Issue alerts notify you when specific conditions are met. To create one:

> 1. Go to **Alerts** and click "Create Alert Rule"

Author: 👍

2. Select "Issues" and click "Set Conditions"
3. Configure "When" conditions (what triggers the alert)
4. Add "If" filters to narrow the scope (optional)
5. Set "Then" actions (where to send notifications)

Common triggers: new issues, high event counts, high user counts, state changes.

Common filters: environment, tags, issue properties.

Learn more about [Creating Alerts](/product/alerts/create-alerts/) and [Issue Alert Configuration](/product/alerts/create-alerts/issue-alert-config/).

## Error Handling Strategy

### Capture Context, Not Just Exceptions

> When you catch an error, add tags, user info, and breadcrumbs to make debugging instant.

Reviewer: I think we want to de-emphasize breadcrumbs, and emphasize replays and logs.

Author: updated in 6a454d6fb

> ```javascript

Author: Yeah it's a little confusing here. Going to de-emphasize breadcrumbs.

Sentry.captureException(error, {
  tags: {
    order_id: order.id,
    payment_gateway: "stripe",
    region: user.region
  },
  user: {
    id: user.id,
    email: user.email
  },
  contexts: {
    order: {
      total: order.total,
      items: order.items.length,
      discount_code: order.discountCode
    }
  }
});
```

**Search in Sentry:** `order_id:order_123` or `payment_gateway:stripe region:us-west`

### Ignore Known Noise

Some errors aren't actionable. Use <PlatformLink to="/configuration/options/#ignore-errors">`ignoreErrors`</PlatformLink> to filter them out.

```javascript
Sentry.init({
  dsn: "...",
  ignoreErrors: [
    "Non-Error promise rejection captured",
    /^Timeout of \d+ms exceeded$/
  ]
});
```

> Also use <PlatformLink to="/configuration/filtering/#using-beforesend">`beforeSend`</PlatformLink> to filter errors dynamically:

Reviewer 2: ```suggestion
You can also use <PlatformLink to="/configuration/filtering/#using-beforesend">`beforeSend`</PlatformLink> to filter errors dynamically:
```
